src/Group.cpp

```diff
@@ -0,0 +1,261 @@
+#include "global.h"
+#include "SongManager.h"
+#include "RageFile.h"
+#include "RageUtil.h"
+#include "RageFileManager.h"
+#include "RageLog.h"
+#include "Song.h"
+#include "SongCacheIndex.h"
+#include "SongUtil.h"
+#include "TitleSubstitution.h"
+#include "Group.h"
+
+#include <cstddef>
+#include <tuple>
+#include <vector>
+
+
+/** @brief The file that contains the group information.
+ * We name this Pack.ini over Group.ini to avoid conflict
+ * with the Waiei Group.ini-lua project still actively
+ * being developed.
+*/
+const RString INI_FILE = "Pack.ini";
+const int INI_VERSION = 1;
+
+Group::Group() {
+    m_sDisplayTitle = "";
+    m_sSortTitle = "";
+    m_sPath = "";
+    m_sGroupName = "";
+    m_sTranslitTitle = "";
+    m_sSeries = "";
+    m_fSyncOffset = 0;
+    m_bHasPackIni = false;
+    m_iYearReleased = 0;
+    m_sBannerPath = "";
+}
+
+Group::~Group() {
+    SONGMAN->GetGroupGroupMap().clear();
+}
+
+Group::Group(const RString& sDir, const RString& sGroupDirName) {
+    RString sPackIniPath = sDir + sGroupDirName + "/" + INI_FILE;
+    RString sDisplayTitle = sGroupDirName;
+    RString sSortTitle = sGroupDirName;
+    RString sTranslitTitle = "";
+    RString sSeries = "";
+    RString sBannerPath = "";
+    float fOffset = PREFSMAN->m_fMachineSyncBias;
+
+    if (FILEMAN->DoesFileExist(sPackIniPath)) {
+        IniFile ini;
+        ini.ReadFile(sPackIniPath);
+        ini.GetValue("Group", "DisplayTitle", sDisplayTitle);
```

Comment:
A lot of this can be cleaned up if you create a vector of all the keys we're interested in.
Like all the string vectors have to go through he "getvalue, trim, check not empty" thing which we do in a loop for all the string fields. The only outlier is the offset which will need special processing.